### ISSUE TITLE
Kernel dies after computing spatial representations

### ISSUE THREAD

**Issue by victormanna** [MICS-Lab/novae#18]
## Kernel dies after computing spatial representations
Hello Quentin, 
Thanks for making Novae
I wanted to reproduce the basic vignette 
https://mics-lab.github.io/novae/tutorials/main_usage/

So I did the following steps
First I installed novae using miniforge
Next,
import novae
adata = novae.utils.load_dataset(tissue="colon", species="human", pattern=".*P2.*")[0]
adata
novae.spatial_neighbors(adata, radius=80)
novae.plot.connectivities(adata)
model = novae.Novae.from_pretrained("MICS-Lab/novae-human-0")
model
Novae model
   ├── Known genes: 60697
   ├── Parameters: 32.0M
   └── Model name: MICS-Lab/novae-human-0

All the steps above works but after running the next line the kernel crashes

## Reproducing the issue
The kernel crashes after completing the computation (after running 10 mins)
#### Option 1: zero-shot
model.compute_representations(adata, zero_shot=True)

OMP: Info #276: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.
Computing representations: 100%|██████████| 666/666 [09:41<00:00,  1.15it/s]
The Kernel crashed while executing code in the current cell or a previous cell. 
Please review the code in the cell(s) to identify a possible cause of the failure. 

I have installed novae in my institute's HPC as per instructions and could again reproduce the same issue.

## System
- Macbook M1 Pro with 16GB RAM
 - OS: [e.g. MacOS 15.3.2]
 - Python version [e.g. 3.10.16]
<details>
<summary>Dependencies versions</summary>
<br>
Package             Version
------------------- -----------
aiohappyeyeballs    2.6.1
aiohttp             3.11.14
aiosignal           1.3.2
anndata             0.11.3
appnope             0.1.4
array_api_compat    1.11.2
asttokens           3.0.0
async-timeout       5.0.1
attrs               25.3.0
certifi             2025.1.31
charset-normalizer  3.4.1
comm                0.2.2
contourpy           1.3.1
cycler              0.12.1
debugpy             1.8.13
decorator           5.2.1
exceptiongroup      1.2.2
executing           2.1.0
filelock            3.18.0
fonttools           4.56.0
frozenlist          1.5.0
fsspec              2025.3.0
h5py                3.13.0
huggingface-hub     0.29.3
idna                3.10
importlib_metadata  8.6.1
ipykernel           6.29.5
ipython             8.34.0
jedi                0.19.2
Jinja2              3.1.6
joblib              1.4.2
jupyter_client      8.6.3
jupyter_core        5.7.2
kiwisolver          1.4.8
legacy-api-wrap     1.4.1
lightning           2.5.1
lightning-utilities 0.14.2
llvmlite            0.44.0
MarkupSafe          3.0.2
matplotlib          3.10.1
matplotlib-inline   0.1.7
mpmath              1.3.0
multidict           6.2.0
natsort             8.4.0
nest_asyncio        1.6.0
networkx            3.4.2
novae               0.2.3
numba               0.61.0
numpy               2.1.3
packaging           24.2
pandas              2.2.3
parso               0.8.4
patsy               1.0.1
pexpect             4.9.0
pickleshare         0.7.5
pillow              11.1.0
pip                 25.0.1
platformdirs        4.3.7
prompt_toolkit      3.0.50
propcache           0.3.0
psutil              7.0.0
ptyprocess          0.7.0
pure_eval           0.2.3
Pygments            2.19.1
pynndescent         0.5.13
pyparsing           3.2.3
python-dateutil     2.9.0.post0
pytorch-lightning   2.5.1
pytz                2025.2
PyYAML              6.0.2
pyzmq               26.3.0
requests            2.32.3
safetensors         0.5.3
scanpy              1.11.0
scikit-learn        1.5.2
scipy               1.15.2
seaborn             0.13.2
session-info2       0.1.2
setuptools          75.8.2
six                 1.17.0
stack_data          0.6.3
statsmodels         0.14.4
sympy               1.13.1
threadpoolctl       3.6.0
torch               2.6.0
torch-geometric     2.6.1
torchmetrics        1.7.0
tornado             6.4.2
tqdm                4.67.1
traitlets           5.14.3
typing_extensions   4.12.2
tzdata              2025.2
umap-learn          0.5.7
urllib3             2.3.0
wcwidth             0.2.13
wheel               0.45.1
yarl                1.18.3
zipp                3.21.0
Note: you may need to restart the kernel to use updated packages.
</details>

**Comment by quentinblampey** [comment]
Thanks @victormanna for reporting, this bug came in the very new version `0.2.3` when working on a single `AnnData` object

I fixed it and directly made a new `novae` version `0.2.4`

Could you please `pip install novae --upgrade` to try the newest version?

**Comment by victormanna** [comment]
Hi Quentin, Many thanks for the fast fix! After updating, the zero shot method works very well on our Xenium 5K dataset (1.1 M cells, 16 samples). It takes around 1h 30 mins on a HPC CPU. On GPU, this takes around 20 mins. Many thanks for developing Novae, this was a much needed tool..
